pkg/backends/jmeter/jmeter.go

```diff
@@ -53,243 +41,35 @@ func New(
-		podAnnotations:       podAnnotations,
-		namespaceAnnotations: namespaceAnnotations,
-		masterResources: helper.Resources{
-			CPULimits:      config.MasterCPULimits,
-			CPURequests:    config.MasterCPURequests,
-			MemoryLimits:   config.MasterMemoryLimits,
-			MemoryRequests: config.MasterMemoryRequests,
-		},
-		workerResources: helper.Resources{
-			CPULimits:      config.WorkerCPULimits,
-			CPURequests:    config.WorkerCPURequests,
-			MemoryLimits:   config.WorkerMemoryLimits,
-			MemoryRequests: config.WorkerMemoryRequests,
-		},
-		masterConfig: loadTestV1.ImageDetails{
-			Image: masterImageName,
-			Tag:   masterImageTag,
-		},
-		workerConfig: loadTestV1.ImageDetails{
-			Image: workerImageName,
-			Tag:   workerImageTag,
-		},
+		backend:   backend,
+		loadTest:  lt,
+		reportURL: reportURL,
 	}
 }
 
 // CheckOrCreateResources check if JMeter kubernetes resources have been create,
 // if they have not been create them
 func (c *JMeter) CheckOrCreateResources(ctx context.Context) error {
-	JMeterServices, err := c.kubeClientSet.CoreV1().Services(c.loadTest.Status.Namespace).List(ctx, metaV1.ListOptions{})
-	if err != nil {
-		return err
-	}
-
-	if len(JMeterServices.Items) == 0 {
-		_, err = c.kubeClientSet.CoreV1().ConfigMaps(c.loadTest.Status.Namespace).Create(ctx, c.NewConfigMap(), metaV1.CreateOptions{})
-		if err != nil && !errors.IsAlreadyExists(err) {
-			c.logger.Error("Error on creating testfile configmap", zap.Error(err))
-			return err
-		}
-
-		_, err = c.kubeClientSet.CoreV1().ConfigMaps(c.loadTest.Status.Namespace).Create(ctx, c.NewJMeterSettingsConfigMap(), metaV1.CreateOptions{})
-		if err != nil && !errors.IsAlreadyExists(err) {
-			return err
-		}
-
-		secret, err := c.NewSecret()
-		if err != nil {
-			c.logger.Error("Error on creating environment variables secret", zap.Error(err))
-			c.loadTest.Status.Phase = loadTestV1.LoadTestErrored
-			return nil
-		}
-
-		_, err = c.kubeClientSet.CoreV1().Secrets(c.loadTest.Status.Namespace).Create(ctx, secret, metaV1.CreateOptions{})
-		if err != nil && !errors.IsAlreadyExists(err) {
-			return err
-		}
-
-		configMaps, err := c.NewTestdataConfigMap()
-		if err != nil {
-			c.logger.Error("Error on creating testdata configMaps", zap.Error(err))
-			c.loadTest.Status.Phase = loadTestV1.LoadTestErrored
-			return nil
-		}
-
-		err = c.createPodsWithTestdata(ctx, configMaps, c.loadTest, c.loadTest.Status.Namespace)
-		if err != nil {
-			return err
-		}
-
-		_, err = c.kubeClientSet.CoreV1().Services(c.loadTest.Status.Namespace).Create(ctx, c.NewJMeterService(), metaV1.CreateOptions{})
-		if err != nil && !errors.IsAlreadyExists(err) {
-			c.logger.Error("Error on creating new JMeter service", zap.Error(err))
-			return err
-		}
-		c.logger.Info(
-			"Created JMeter resources",
-			zap.String("LoadTest", c.loadTest.GetName()),
-			zap.String("namespace", c.loadTest.Status.Namespace),
-		)
-
-	}
-	return nil
+	return c.backend.Sync(ctx, *c.loadTest, c.reportURL)
```

Comment:
I would also add debug logging on each `CheckOrCreateResources` or we can move `debug` log to `Sync` method.

Reply:
which log are you referring to? I didn’t remove any log, or did?

Reply:
I think we didn't have any. I thought maybe as addition, might be useful when you debugging controller.

Reply:
Sounds like a good idea for another Ticket + PR